src/Microsoft.AspNetCore.Server.Kestrel.Core/Features/IHttpRequestBodyTimeoutFeature.cs

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Core.Features
+{
+    /// <summary>
+    /// Represents a timeout for the request body in an HTTP request.
+    /// </summary>
+    public interface IHttpRequestBodyTimeoutFeature
+    {
+        /// <summary>
+        /// The maximum amount of time in which the request body should be fully received.
+        /// </summary>
+        TimeSpan RequestBodyTimeout { get; set; }
```

Comment:
Up to a limit. You give it a range (e.g. 20-40 seconds) and it will increase the timeout by 1 second up to the upper bound as long as data is arriving at the specified rate.